OuterWildsRandomSpeedrun/SpawnPointMenu/SpawnPointSelectorManager.cs

```diff
@@ -0,0 +1,200 @@
+using OWML.Common;
+using SpawnPointSelector;
+using UnityEngine;
+using System;
+using System.Collections.Generic;
+using static UnityEngine.InputSystem.InputAction;
+
+namespace OuterWildsRandomSpeedrun
+{
+  public class SpawnPointSelectorManager : MonoBehaviour
+  {
+    public IModHelper ModHelper { get; set; }
+
+    public static SpawnPointSelectorManager Instance {
+      get => _instance;
+    }
+
+    private bool _menuDisplayed = false;
+
+    /// <summary>
+    /// The top-level script associated with the spawn selector UI
+    /// </summary>
+    private SpawnPointSelector.SpawnPointSelector _selector;
+
+    /// <summary>
+    /// The "From" menu, represented in Outer Wild's format
+    /// </summary>
+    private SpawnPointMenu _fromMenu;
+
+    /// <summary>
+    /// The "To" menu, represented in Outer Wild's format
+    /// </summary>
+    private SpawnPointMenu _toMenu;
+
+    /// <summary>
+    /// The "From" menu, represented as a script associated with our prefab
+    /// </summary>
+    private SpawnPointList _fromList;
+
+    /// <summary>
+    /// The "To" menu, represented as a script associated with our prefab
+    /// </summary>
+    private SpawnPointList _toList;
+
+    private static SpawnPointSelectorManager _instance;
+
+    public void Awake()
+    {
+      _instance = this;
+    }
+
+    public void DisplayMenu()
+    {
+      _menuDisplayed = true;
+
+      InitializeSelector();
+      InitializeMenus();
+
+      _selector.gameObject.SetActive(true);
+      
+      var action = (InputLibrary.menuLeft as InputCommands).Action as BasicInputAction;

```

Comment:
name `action` something more specific, Unity has Action already (as indicated in this line haha)

Reply:
done